### PR TITLE
Add redo command

### LINKED ISSUE #113: Redo command
As a user, I can redo commands after undoing them, so that I would not have to manually type out the command again.

### PR DESCRIPTION
**Pull request by bryanyee33**
Closes #113

### CODE REVIEW

File: src/main/java/seedu/address/model/AddressBook.java [hunk 1]
```diff
@@ -33,15 +40,32 @@ public AddressBook(ReadOnlyAddressBook toBeCopied) {
...
 
-    //// list overwrite operations
+    //// undo & redo operations
 
     /**
-     * Makes a copy of persons, and stores it in personsList.
+     * Empties the redoList.
      */
-    private void save() {
+    private void emptyRedoList() {
```
Comment (aureliony) [inline src/main/java/seedu/address/model/AddressBook.java:48]: Would `clearRedoList` be a better name for this method?

**Review by aureliony — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/model/AddressBook.java (hunk 1 above)
Comment (bryanyee33) [reply src/main/java/seedu/address/model/AddressBook.java]: Yes, it has been changed.

File: src/main/java/seedu/address/model/AddressBook.java [hunk 2]
```diff
@@ -33,34 +40,73 @@ public AddressBook(ReadOnlyAddressBook toBeCopied) {
...
 
     /**
-     * Makes a copy of persons, and stores it in personsList.
+     * Empties the redoList.
      */
-    private void save() {
+    private void clearRedoList() {
+        while (!redoStack.empty()) {
+            redoStack.pop();
+        }
```
Comment (aureliony) [inline src/main/java/seedu/address/model/AddressBook.java:51]: This method body can be replaced with `redoStack.clear()`.

File: src/main/java/seedu/address/model/AddressBook.java [hunk 3]
```diff
@@ -33,34 +40,73 @@ public AddressBook(ReadOnlyAddressBook toBeCopied) {
...
 
-    //// list overwrite operations
+    //// undo & redo operations
 
     /**
-     * Makes a copy of persons, and stores it in personsList.
+     * Empties the redoList.
      */
-    private void save() {
+    private void clearRedoList() {
```
Comment (aureliony) [inline src/main/java/seedu/address/model/AddressBook.java:48]: This method seems to be redundant as there is only one call that directly use `redoStack.clear()`.

**Review by aureliony — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/model/AddressBook.java (hunk 2 above)
Comment (bryanyee33) [reply src/main/java/seedu/address/model/AddressBook.java]: Nice find! It was absent in the javadoc website so I didn't know it existed.

**Review by aureliony — APPROVED** [review]

**Pull request merged**